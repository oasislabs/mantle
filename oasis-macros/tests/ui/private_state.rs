oasis_std::contract! {

#[derive(Contract)]
struct Counter(u32);

impl Counter {
    pub fn new(ctx: Context) -> Self {
        Self(42)
    }
}

}

fn main() {}
