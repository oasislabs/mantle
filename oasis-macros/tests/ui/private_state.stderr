error: `struct Counter` should have `pub` visibility.
 --> $DIR/private_state.rs:4:1
  |
4 | struct Counter(u32);
  | ^^^^^^

error: aborting due to previous error

